app/src/main/java/com/alfresco/content/app/activity/ExtensionActivity.kt

```diff
@@ -26,37 +33,40 @@ class ExtensionActivity : AppCompatActivity(), MavericksView {
     private val navController by lazy { findNavController(R.id.nav_host_fragment) }
     private lateinit var actionBarController: ActionBarController
     private var signedOutDialog = WeakReference<AlertDialog>(null)
+    private var internetUnavailableDialog = WeakReference<AlertDialog>(null)
+    private val shareLimit = 50
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_extension)
 
-        when (intent?.action) {
-            Intent.ACTION_SEND -> {
-                println("EXTENSION SINGLE IMAGE")
-                if (intent.type?.startsWith("image/") == true) {
-                    handleSendImage(intent)
+        intent?.let { intentObj ->
+            if (intentObj.hasExtra(EXTRA_IS_LOGIN) && intentObj.getBooleanExtra(EXTRA_IS_LOGIN, false)) {
+                configure()
+                return
+            } else {
+                when (intent?.action) {
+                    Intent.ACTION_SEND -> {
+                        handleSendImage(intent)
+                    }
+                    Intent.ACTION_SEND_MULTIPLE -> {
```

Comment:
Why there are two different functions to handle images?

Reply:
Because if you have single file then it received in ACTION_SEND otherwise ACTION_SEND_MULTIPLE